crates/orchestrator/src/network/node.rs

```diff
@@ -37,6 +37,58 @@ pub struct NetworkNode {
     metrics_cache: Arc<RwLock<MetricMap>>,
 }
 
+/// Result of waiting for a certain number of log lines to appear.
+///
+/// Indicates whether the log line count condition was met within the timeout period.
+///
+/// # Variants
+/// - `TargetReached(count)` – The predicate condition was satisfied within the timeout.
+///     * `count`: The number of matching log lines at the time of satisfaction.
+/// - `TargetFailed(count)` – The condition was not met within the timeout.
+///     * `count`: The final number of matching log lines at timeout expiration.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LogLineCount {
+    TargetReached(u32),
+    TargetFailed(u32),
+}
+
+/// Configuration for controlling log line count waiting behavior.
+///
+/// Allows specifying a custom predicate on the number of matching log lines,
+/// a timeout in seconds, and whether the system should wait the entire timeout duration.
+///
+/// # Fields
+/// - `predicate`: A function that takes the current number of matching lines and
+///   returns `true` if the condition is satisfied.
+/// - `timeout_secs`: Maximum number of seconds to wait.
+/// - `wait_until_timeout_elapses`: If `true`, the system will continue waiting
+///   for the full timeout duration, even if the condition is already met early.
+///   Useful when you need to verify sustained absence or stability (e.g., "ensure no new logs appear").
+#[derive(Clone)]
+pub struct LogLineCountOptions {
+    pub predicate: Arc<dyn Fn(u32) -> bool + Send + Sync>,
+    pub timeout_secs: u64,
```

Comment:
nit: IMO this timeout should be a `Duration`. Makes the API a bit nicer.

Reply:
Will do :)

Reply:
The `new` method accept `impl Into<u64>` but we can change the type to be a `Duration` for better _dx_

Thanks!